app/src/main/kotlin/com/wire/android/ui/userprofile/UserProfileViewModel.kt

```diff
@@ -57,10 +58,41 @@ class UserProfileViewModel @Inject constructor(private val navigationManager: Na
     }
 
     fun editProfile() {
-        //TODO
+        viewModelScope.launch {
+            navigationManager.navigate(NavigationCommand(NavigationItem.Settings.route))
+        }
+    }
+
+    fun onDismissDialog() {
+        userProfileState = userProfileState.copy(dialogState = DialogState.None)
     }
 
     fun changeStatus(status: UserStatus) {
         //TODO
+        onDismissDialog()
+    }
+
+    fun doNotShowStatusDialogCheckChanged(doNotShow: Boolean) {
+        userProfileState.run {
+            if (dialogState is DialogState.StatusChange) {
+                userProfileState = copy(dialogState = dialogState.changeCheckBoxState(doNotShow))
```

Comment:
This is not really persisting the "Don't show again status dialog" checkbox state